### PR TITLE
Deploying qa-brh.planx-pla.net - 2022.09 1665512168

### PR DESCRIPTION
**Pull request by PlanXCyborg**
Applying version 2022.09 to qa-brh.planx-pla.net

### CODE REVIEW

**Comment by PlanXCyborg** [comment]
# qa-brh.planx-pla.net/manifest.json
## Deployment changes
- arborist
  - Replace Travis with a GitHub workflow for CI (uc-cdis/arborist#150)
- fence
  - Configure the `ALLOWED_DATA_UPLOAD_BUCKETS` setting to allow users to upload to buckets other than `DATA_UPLOAD_BUCKET` (uc-cdis/fence#1051)
  - B/c the default session expiration changed from 30 minutes to 15 minutes to comply with FedRAMP, you MUST ensure that an instance that should comply with FedRAMP does NOT override this fence configuration (e.g. it should be 15 minutes) `SESSION_TIMEOUT: 900` (uc-cdis/fence#1040)
  - Requires a Fence DB migration (`Client.redirect_uri` is now optional) (uc-cdis/fence#1033)
  - Requires database migration for a new table to Fence, google Access expiration to add new `expires` column, and to create iss_sub_pair_to_user table #968  #963  #973 (uc-cdis/fence#964)
  - Requires Arborist >= `2022.04` OR >=`3.4.0` https://github.com/uc-cdis/arborist/pull/143 (uc-cdis/fence#964)
  - To enable Passports -> DRS in an environment you must update Fence Configuration to at a minimum specify `GA4GH_PASSPORTS_TO_DRS_ENABLED: true` (**only do this after coordinating with the environment owner as this has several requirements before it can be enabled, including the need to use Indexd's `authz` field in ALL records rather than `acl`**) (uc-cdis/fence#964)
## Breaking changes
- fence
  - Default session expiration changed from 30 minutes to 15 minutes to comply with FedRAMP (uc-cdis/fence#1040)
  - URL Signing when no_force_sign query param is provided: Previously Fence would make a decision based off if the data was public and no_force_sign provided. Fence will now NEVER sign if no_force_sign is provided (since the concept of "public" data has been abstracted out of Fence. Data access - public or not - is the sole responsibility of the policy engine). In other words, if no_force_sign is provided at the API level, Fence will respect that regardless of whether the resulting URL will actually work. The default Fence behavior should remain the same. #988 (uc-cdis/fence#964)
- indexd
  - Ensure you are using Fence>=2021.10. Remove deprecated GA4GH access endpoints. these have existed (and been used) from the Fence microservice since last year. See cloud automation change [here](https://github.com/uc-cdis/cloud-automation/blob/f10f04f7957eb2ba179504f7df9d6429f7298835/gen3/bin/kube-setup-revproxy.sh#L78-L79) (uc-cdis/indexd#336)

File: qa-brh.planx-pla.net/manifests/hatchery/hatchery.json [hunk 1]
```diff
@@ -11,7 +11,7 @@
   "sidecar": {
     "cpu-limit": "1.0",
     "memory-limit": "256Mi",
-    "image": "quay.io/cdis/ecs-ws-sidecar:feat_datamounting",
+    "image": "quay.io/cdis/ecs-ws-sidecar:2022.09",
```
Comment (aartivnkt) [inline qa-brh.planx-pla.net/manifests/hatchery/hatchery.json:14]: not sure if `feat_datamounting` image is needed for hatchery. @jmontmaxwell do you know about this?

**Review by nss10 — DISMISSED** [review]

File: qa-brh.planx-pla.net/manifests/hatchery/hatchery.json (hunk 1 above)
Comment (jmontmaxwell) [reply qa-brh.planx-pla.net/manifests/hatchery/hatchery.json:14]: Looking at the ecs-ws-sidecar history it looks like feat_datamounting was merged in May2022 so I think we should be fine @aartivnkt

Comment (atharvar28) [reply qa-brh.planx-pla.net/manifests/hatchery/hatchery.json:14]: 👍

**Comment by PlanXCyborg** [comment]
# qa-brh.planx-pla.net/manifest.json
## Deployment changes
- arborist
  - Replace Travis with a GitHub workflow for CI (uc-cdis/arborist#150)
- fence
  - Configure the `ALLOWED_DATA_UPLOAD_BUCKETS` setting to allow users to upload to buckets other than `DATA_UPLOAD_BUCKET` (uc-cdis/fence#1051)
  - B/c the default session expiration changed from 30 minutes to 15 minutes to comply with FedRAMP, you MUST ensure that an instance that should comply with FedRAMP does NOT override this fence configuration (e.g. it should be 15 minutes) `SESSION_TIMEOUT: 900` (uc-cdis/fence#1040)
  - Requires a Fence DB migration (`Client.redirect_uri` is now optional) (uc-cdis/fence#1033)
  - Requires database migration for a new table to Fence, google Access expiration to add new `expires` column, and to create iss_sub_pair_to_user table #968  #963  #973 (uc-cdis/fence#964)
  - Requires Arborist >= `2022.04` OR >=`3.4.0` https://github.com/uc-cdis/arborist/pull/143 (uc-cdis/fence#964)
  - To enable Passports -> DRS in an environment you must update Fence Configuration to at a minimum specify `GA4GH_PASSPORTS_TO_DRS_ENABLED: true` (**only do this after coordinating with the environment owner as this has several requirements before it can be enabled, including the need to use Indexd's `authz` field in ALL records rather than `acl`**) (uc-cdis/fence#964)
## Breaking changes
- fence
  - Default session expiration changed from 30 minutes to 15 minutes to comply with FedRAMP (uc-cdis/fence#1040)
  - URL Signing when no_force_sign query param is provided: Previously Fence would make a decision based off if the data was public and no_force_sign provided. Fence will now NEVER sign if no_force_sign is provided (since the concept of "public" data has been abstracted out of Fence. Data access - public or not - is the sole responsibility of the policy engine). In other words, if no_force_sign is provided at the API level, Fence will respect that regardless of whether the resulting URL will actually work. The default Fence behavior should remain the same. #988 (uc-cdis/fence#964)
- indexd
  - Ensure you are using Fence>=2021.10. Remove deprecated GA4GH access endpoints. these have existed (and been used) from the Fence microservice since last year. See cloud automation change [here](https://github.com/uc-cdis/cloud-automation/blob/f10f04f7957eb2ba179504f7df9d6429f7298835/gen3/bin/kube-setup-revproxy.sh#L78-L79) (uc-cdis/indexd#336)

**Review by jmontmaxwell — APPROVED** [review]

**Review by aartivnkt — APPROVED** [review]

**Pull request merged**